src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -776,27 +776,25 @@ private void Get_CORINFO_SIG_INFO(MethodDesc method, CORINFO_SIG_INFO* sig, Meth
 
                 if (method.IsArrayAddressMethod())
                     hasHiddenParameter = true;
-
-                // We only populate sigInst for intrinsic methods because most of the time,
-                // JIT doesn't care what the instantiation is and this is expensive.
-                Instantiation owningTypeInst = method.OwningType.Instantiation;
-                sig->sigInst.classInstCount = (uint)owningTypeInst.Length;
-                if (owningTypeInst.Length != 0)
-                {
-                    sig->sigInst.classInst = GetJitInstantiation(owningTypeInst);
-                }
-
-                sig->sigInst.methInstCount = (uint)method.Instantiation.Length;
-                if (method.Instantiation.Length != 0)
-                {
-                    sig->sigInst.methInst = GetJitInstantiation(method.Instantiation);
-                }
             }
 
             if (hasHiddenParameter)
             {
                 sig->callConv |= CorInfoCallConv.CORINFO_CALLCONV_PARAMTYPE;
             }
+
+            Instantiation owningTypeInst = method.OwningType.Instantiation;
+            sig->sigInst.classInstCount = (uint)owningTypeInst.Length;
+            if (owningTypeInst.Length != 0)
+            {
+                sig->sigInst.classInst = GetJitInstantiation(owningTypeInst);
+            }
+
+            sig->sigInst.methInstCount = (uint)method.Instantiation.Length;
+            if (method.Instantiation.Length != 0)
+            {
+                sig->sigInst.methInst = GetJitInstantiation(method.Instantiation);
+            }
```

Comment:
I have moved this out of the `method.IsIntrinsic` path to unconditionally fill it, since it is used by JIT for `JitDisasm` now and making assumptions on the exact scenarios this is used by the JIT is a bit icky.

If we are worried about performance (I did not see any measurable difference in crossgen2 times of SPC) then I think we should remove `CORINFO_SIG_INFO::sigInst` and put this on the same plan as `ICorStaticInfo::getTypeInstantiationArgument`. But I think we can wait and see the crossgen2 TP graphs in PowerBI.